pkg/services/alerting/conditions/query.go

```diff
@@ -360,3 +358,19 @@ func FrameToSeriesSlice(frame *data.Frame) (tsdb.TimeSeriesSlice, error) {
 
 	return seriesSlice, nil
 }
+
+func toCustomError(err error) error {
+	// is context timeout
+	if errors.Is(err, gocontext.DeadlineExceeded) {
+		return fmt.Errorf("alert execution exceeded the timeout")
+	}
+
+	// is Prometheus error
+	var e prometheus.PrometheusError
+	if errors.As(err, &e) {
```

Comment:
You're not doing anything with the converted error, so should be `errors.Is`
```suggestion
	if errors.Is(err, prometheus.PrometheusError) {
```

Reply:
I tried this pattern but it didn't work as I expected. It's not defined with `errors.New`, just given the `error` type so I'm not sure the suggestion applies?

Reply:
I think you need to create a struct rather than an alias type: https://play.golang.org/p/Ak4SLvQ2APB

Reply:
Thanks for the tip @simonswine - made the wrong assumption that alias represented a distinct type. Have reworked it a different way without introducing an error type